config/v1/types_proxy.go

```diff
@@ -14,9 +14,11 @@ type Proxy struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 	// Spec holds user-settable values for the proxy configuration
 	// +required
-	Spec ProxySpec `json:"spec"`
+	Spec   ProxySpec   `json:"spec"`
+	Status ProxyStatus `json:"status"`
 }
 
+// ProxySpec contains desired cluster proxy configuration
```

Comment:
nit: godocs [should end in periods][1].

[1]: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences